controllers/cloudstackcluster_controller.go

```diff
@@ -21,6 +21,8 @@ import (
 	"reflect"
 
 	"github.com/go-logr/logr"
+	"github.com/hashicorp/go-multierror"
```

Comment:
We had a few issues in error handling patches that made troubleshooting the endpoint/webhook issue difficult.

Instead of getting tricky about returning errors, I instead use multierror and return both errors.